packages/belos/src/BelosBlockGmresSolMgr.hpp

```diff
@@ -580,28 +578,30 @@ void BlockGmresSolMgr<ScalarType,MV,OP>::setParameters( const Teuchos::RCP<Teuch
   // Check if the orthogonalization changed.
   if (params->isParameter("Orthogonalization")) {
     std::string tempOrthoType = params->get("Orthogonalization",orthoType_default_);
+#ifdef HAVE_BELOS_TSQR
+    TEUCHOS_TEST_FOR_EXCEPTION( tempOrthoType != "DGKS" && tempOrthoType != "ICGS" && tempOrthoType != "IMGS" && tempOrthoType != "TSQR",
```

Comment:
Don't check the orthogonalization name here; the `OrthoManagerFactory` will check input.  That will avoid scattering `#ifdef HAVE_BELOS_TSQR ... #endif` in many places throughout the code.  We can delete all of these checks (down to and including line 589).